### ISSUE TITLE
Hostname dhcp registration

### ISSUE THREAD

**Issue by Sithuk** [zbm-dev/zfsbootmenu#285]
I'd haven't been able to remotely connect to zfsbootmenu using its hostname and the local network domain (FQDN addressing). Connecting by IP works. It looks like the "ZFSBootMenu" hostname isn't registered with the dhcp server.

ahesford recommended adding the following parameters directly to the zfsbootmenu  image in the remote access wiki guide, which is what I have done.
https://github.com/zbm-dev/zfsbootmenu/wiki/Remote-Access-to-ZBM
ip=dhcp rd.neednet=1

The following commit added a hostname of "ZFSBootMenu" in zfsbootmenu v.1.10
https://github.com/zbm-dev/zfsbootmenu/commit/64bd359687351528545230af6847c68d4b0b84d8

Unfortunately the hostname doesn't appear to be registered with the dhcp server. An IP address is assigned but the hostname isn't registered.

I was hoping to be able to connect using hostname DNS addressing and avoid having to specify a static ip. For example, connect with: "ssh root@ZFSBootMenu.lan -p 222". I've tested using a manual override to define the hostname using the following content in /etc/cmdline.d/dracut-network.conf.
ip=::::ZFSBootMenu::dhcp rd.neednet=1

ZFSbootmenu connects and configures automatically using dhcp, but the hostname isn't registered. So no different to just using "ip=dhcp rd.neednet=1"

The same issue seems to be the same as reported here:
https://bugs.centos.org/view.php?id=12828

Have you been able to get hostname DNS addressing to work with the ZBM remote access feature?

**Comment by ahesford** [comment]
Making this work requires that your DHCP server honor hostnames sent by the client, and a client that will send a hostname to the server. `dhcpcd` can do this; `dhclient` may be able to as well. Either way, you're in for a lot of configuration that is probably more trouble than it's worth. You're better off reserving IP addresses for machines you want to unlock remotely.

**Comment by Sithuk** [comment]
Thanks for the quick response. I'll use static addressing for the Ubuntu Zfsbootmenu install script and offer dhcp as an option.

Just so I understand, is the issue that dracut doesn't send a hostname to the server, and as zfsbootmenu uses dracut, then the hostname isn't sent? (My DHCP server honours hostnames sent by the client)

**Comment by ahesford** [comment]
The issue isn't really dracut, it's the DHCP client that the dracut network module installs. The module installs dhclient, and I'm not sure how well dhclient works for this kind of thing. If you build an image in "hostonly" mode, it will copy your `/etc/dhclient.conf` into the initramfs, then it appends its own configuration. You *might* have success if you add `send host-name "whatever-you-like";` to `/etc/dhclient.conf` and make sure you specify `hostonly=yes` in your dracut configuration.

**Comment by Sithuk** [comment]
Thank you ahesford, using your advice I've been able to solve the issue. The relevant file was:
/usr/lib/dracut/modules.d/35network-legacy/dhclient.conf

I added: 
send host-name "whatever-you-like";

Then generate-zbm to update the initramfs.

I've updated the Ubuntu zfsbootmenu install script, which I've recently relocated to github. Thank you again.
https://github.com/Sithuk/ubuntu-server-zfsbootmenu
